api/types/clusterconfig.go

```diff
@@ -102,6 +102,15 @@ type ClusterConfig interface {
 
 	// Copy creates a copy of the resource and returns it.
 	Copy() ClusterConfig
+
+	// ApplyOverride applies overrides specified in ClusterConfigOverride.
+	ApplyOverride(ClusterConfigOverride) error
+
+	// ExtractOverride returns the currently applied override.
+	ExtractOverride() (ClusterConfigOverride, error)
```

Comment:
Nit: I'd name this `GetOverride()` to be consistent with other `GetXXX()` methods.

Reply:
This is not a genuine getter though. It does not simply return an internal property, it builds a new object and can fail. Wouldn't it be confusing to use the naming convention for getters?

Reply:
Yeah that's fair. We can keep it as-is.